wgsl/index.bs

```diff
@@ -824,6 +826,81 @@ See [[#arithmetic-expr]].
   </xmp>
 </div>
 
+### Atomic Types ### {#atomic-types}
+
+An <dfn noexport>atomic type</dfn> encapsulates a [=scalar=] type such that:
+* atomic objects provide certain guarantees to concurrent observers, and
+* the only valid operations on atomic objects are the [[#atomic-builtin-functions|atomic builtin functions]].
+
+<table class='data'>
+  <thead>
+    <tr><th>Type<th>Description
+  </thead>
+  <tr algorithm="atomic type"><td>atomic&lt;|T|&gt;
+    <td>Atomic of type |T|. |T| must be either [=u32=] or [=i32=].
+</table>
+
+An expression must not evaluate to an atomic type.
```

Comment:
On the face of it, this is a bit surprising. I think it's correct, but should probably be clarified (possibly by mentioning the difference between ref types vs non-ref types)

Reply:
We use the kind of wording to disallow expressions that evaluate to a runtime-sized array.

Yes, it does rely on the subtle interaction with the load rule.

Yes, I can follow up with a PR that modifies both spots.